geomstats/learning/natural_gradient_descent.py

```diff
@@ -0,0 +1,110 @@
+"""Natural gradient descent using Amari's anaytic solution.
+
+References
+----------
+.. [AS1985] Amari S., Karakida R., Oizumi M. (2018)
+    Fisher Information and Natural Gradient Learning of Random Deep Networks.
+"""
+
+import math
+import random
+
+import numpy as np
+import torch
+from torch.optim import Optimizer
+
+import geomstats.backend as gs
+
+
+class NaturalGradientDescent(Optimizer):
+    """Natural Gradient Descent.
+
+    Uses Amari's analytic solution for the inverse Fisher information matrix.
+
+    Parameters
+    ----------
+    params : array-like, shape=[3, dim]
+        The current layer of the ResNet, consisting of a vector of nodes,
+        a vector of weights, and a bias scalar
+    lr : float
+        The learning rate to update the weights
+    wd : float
+        Penalization on weights to prevent overfitting
+    bias : Boolean
+        Whether or not to include bias term in training
+
+    References
+    ----------
+    .. [AS1985] Amari S., Karakida R., Oizumi M. (2018)
+        Fisher Information and Natural Gradient Learning of Random Deep Networks.
+    """
+
+    def __init__(self, params, lr=0.05, wd=0.0, bias=True):
+        defaults = dict(lr=lr, wd=wd, bias=bias)
+        self.params = params
+
+        super().__init__(params, defaults)
+
+    def step(self):
+        """Performs one step of Natural Gradient Descent"""
+        w_list = []  
+        w_star_list = [] 
+        for group in self.param_groups: # only 1 group for now
+            for i, param in enumerate(group['params']):
+                if i % 2 == 0: # only want to capture the weight parameters
+                    param2 = param.detach().numpy()
+                    w_list.append(param2)
+                    # Append each bias value to the corresponding weight vector
+                    bias_param = group["params"][i + 1]
+                    param_concat_bias = torch.cat(
+                        (param.data, bias_param.view(-1, 1)), dim=1
+                    )
+                    param2 = param_concat_bias.detach().numpy()
+                    w_star_list.append(param2)
+
+            for layer_index, weight_layer in enumerate(w_list):
+                A_00_list = []
```

Comment:
For readability to future users/contributors, let's make the variable names understandable. 
In particular:
- no uppercase letters for variables, unless they are classes.
As per python coding style guidelines: https://peps.python.org/pep-0008/
- to "single letter" names, instead, prefer english names.